bccsp/pkcs11/pkcs11.go

```diff
@@ -84,25 +84,42 @@ func loadLib(lib, pin, label string) (*pkcs11.Ctx, uint, *pkcs11.SessionHandle,
 func (csp *impl) getSession() (session pkcs11.SessionHandle) {
 	select {
 	case session = <-csp.sessions:
-		logger.Debugf("Reusing existing pkcs11 session %+v on slot %d\n", session, csp.slot)
+		_, err := csp.ctx.GetSessionInfo(session)
```

Comment:
I had mentioned to Saad that  it is possible for the Session to be opened, but to be in an invalid state. So it's not enough to simply call this function. The SessionInfo object returns a DeviceError and State code. When State has a code of 0 (I believe, but I'll double check the spec), the session itself is in an invalid state, and then we also need to eject the session from the pool.

Reply:
Thanks Brett. When we checked, any error in the session would result in an error response. Please let us know if it is otherwise.

As per the session state, following are the states that it can be:
```
CKS_RO_PUBLIC_SESSION
CKS_RO_USER_FUNCTIONS
CKS_RW_PUBLIC_SESSION
CKS_RW_USER_FUNCTIONS
CKS_RW_SO_FUNCTIONS 
```

For the full info response, search for "CK_SESSION_INFO" in [this document](https://docs.oasis-open.org/pkcs11/pkcs11-base/v2.40/os/pkcs11-base-v2.40-os.html)

Reply:
You are right and agreed, the other consideration is section 5.1.1:  CKR_FUNCTION_FAILED which states that even if the ulDeviceError field contains an error code, its likely a subsequent call to the same function would succeed. So I'll retract my comment.

Thanks Mihir for linking to the spec.